x-pack/plugins/rule_registry/server/utils/create_persistence_rule_type_wrapper.ts

```diff
@@ -160,17 +164,34 @@ export const createPersistenceRuleTypeWrapper: CreatePersistenceRuleTypeWrapper
                   return { createdAlerts: [], errors: {}, alertsWereTruncated };
                 }
 
-                return {
-                  createdAlerts: augmentedAlerts
-                    .map((alert, idx) => {
-                      const responseItem = response.body.items[idx].create;
-                      return {
-                        _id: responseItem?._id ?? '',
-                        _index: responseItem?._index ?? '',
-                        ...alert._source,
-                      };
+                const createdAlerts = augmentedAlerts
+                  .map((alert, idx) => {
+                    const responseItem = response.body.items[idx].create;
+                    return {
+                      _id: responseItem?._id ?? '',
+                      _index: responseItem?._index ?? '',
+                      ...alert._source,
+                    };
+                  })
+                  .filter((_, idx) => response.body.items[idx].create?.status === 201)
+                  // Security solution's EQL rule consists of building block alerts which should be filtered out.
+                  // Building block alerts have additional "kibana.alert.group.index" attribute which is absent for the root alert.
+                  .filter((alert) => !Object.keys(alert).includes(ALERT_GROUP_INDEX));
+
+                createdAlerts.forEach((alert) =>
+                  options.services.alertFactory
+                    .create(alert._id)
+                    .scheduleActions(type.defaultActionGroupId, {
+                      rule: mapKeys(snakeCase, {
+                        ...options.params,
+                        name: options.rule.name,
+                        id: options.rule.id,
```

Comment:
do we want to make the alert available in the context variable? or a result_link?

Reply:
Yeah I'm not seeing alert data available in `contxext.alerts` for the per alert actions - do you have to use a different mustache variable for per alert actions to get the alert data? If not, we should probably add it in `context.alerts`.

Reply:
I intentionally removed `context.alerts` for per alert actions [here](https://github.com/elastic/kibana/pull/155384/files#diff-c718bc61f925a15ede964cff08fcaf559a2cfe86f313be889867876faebddb81R444). I was not sure whether we need it for the case when we have one alert only. I will add it back and make sure we pass `alerts` for single alert case as well, which will be just `alerts: [alert]`.

Reply:
@ymao1 @marshallmain I added `results_link: alert[ALERT_URL]` for per alert action. According to [these changes](https://github.com/elastic/kibana/pull/155069/files#diff-2ce4793aca22f29f47b8bf684acf32d36ae80ada0aaad095fec72549fbb1d8d7R28) `alert[ALERT_URL]` can be `undefined`. What would be a good alternative for the `results_link` in this case?

Reply:
Hey @e40pud it can be undefined because it's based on the `server.publicBaseUrl` being set. We expect users to have that configured as connectors apparently won't work without it, but there's a chance they may not always have it configured. I'm not sure about what a smart alternative default would be

Reply:
Ya, there is no good alternative.  In similar cases in alerting, we use `undefined`.  It's possible to "if" this condition (falsy) in mustache, so customers _can_ "mustache" it in or out, if it's there or not.